Tests/UnitTests/Core/TrackFitting/Gx2fTests.cpp

```diff
@@ -0,0 +1,352 @@
+
+using namespace Acts::UnitLiterals;
+
+namespace Acts {
+namespace Test {
+
+//// Construct initial track parameters.
+Acts::CurvilinearTrackParameters makeParameters(
+    const ActsScalar x = 0.0_m, const ActsScalar y = 0.0_m,
+    const ActsScalar z = 0.0_m, const ActsScalar w = 42_ns,
+    const ActsScalar phi = 0_degree, const ActsScalar theta = 90_degree,
+    const ActsScalar p = 1_GeV, const ActsScalar q = 1_e) {
+  // create covariance matrix from reasonable standard deviations
+  Acts::BoundVector stddev;
+  stddev[Acts::eBoundLoc0] = 100_um;
+  stddev[Acts::eBoundLoc1] = 100_um;
+  stddev[Acts::eBoundTime] = 25_ns;
+  stddev[Acts::eBoundPhi] = 2_degree;
+  stddev[Acts::eBoundTheta] = 2_degree;
+  stddev[Acts::eBoundQOverP] = 1 / 100_GeV;
+  Acts::BoundSymMatrix cov = stddev.cwiseProduct(stddev).asDiagonal();
+  // define a track in the transverse plane along x
+  Acts::Vector4 mPos4(x, y, z, w);
+  return Acts::CurvilinearTrackParameters(mPos4, phi, theta, p, q, cov);
+}
+
+// Construct a straight-line propagator.
+auto makeStraightPropagator(std::shared_ptr<const Acts::TrackingGeometry> geo) {
+  Acts::Navigator::Config cfg{std::move(geo)};
+  cfg.resolvePassive = false;
+  cfg.resolveMaterial = true;
+  cfg.resolveSensitive = true;
+  Acts::Navigator navigator(cfg);
+  Acts::StraightLineStepper stepper;
+  return Acts::Propagator<Acts::StraightLineStepper, Acts::Navigator>(
+      stepper, std::move(navigator));
+}
+
+static std::vector<Acts::SourceLink> prepareSourceLinks(
+    const std::vector<TestSourceLink>& sourceLinks) {
+  std::vector<Acts::SourceLink> result;
+  std::transform(sourceLinks.begin(), sourceLinks.end(),
+                 std::back_inserter(result),
+                 [](const auto& sl) { return Acts::SourceLink{sl}; });
+  return result;
+}
+
+std::shared_ptr<const TrackingGeometry> makeToyDetector(
+    const GeometryContext& tgContext, const size_t nSurfaces = 5) {
+  if (nSurfaces < 1) {
+    throw std::invalid_argument("At least 1 surfaces needs to be created.");
+  }
+  // Construct builder
+  CuboidVolumeBuilder cvb;
+
+  // Create configurations for surfaces
+  std::vector<CuboidVolumeBuilder::SurfaceConfig> surfaceConfig;
+  for (unsigned int i = 1; i <= nSurfaces; i++) {
+    // Position of the surfaces
+    CuboidVolumeBuilder::SurfaceConfig cfg;
+    cfg.position = {i * UnitConstants::m, 0, 0.};
+
+    // Rotation of the surfaces
+    double rotationAngle = M_PI * 0.5;
+    Vector3 xPos(cos(rotationAngle), 0., sin(rotationAngle));
+    Vector3 yPos(0., 1., 0.);
+    Vector3 zPos(-sin(rotationAngle), 0., cos(rotationAngle));
+    cfg.rotation.col(0) = xPos;
+    cfg.rotation.col(1) = yPos;
+    cfg.rotation.col(2) = zPos;
+    /// Shape of the surface
+    // Boundaries of the surfaces
+    cfg.rBounds =
+        std::make_shared<const RectangleBounds>(RectangleBounds(0.5_m, 0.5_m));
+
+    // Material of the surfaces
+    MaterialSlab matProp(makeBeryllium(), 0.5_mm);
```

Comment:
Just out of curiosity: why Beryllium?

Reply:
There is no strong reason behind Beryllium. We use Be and Si almost as often, since the CuboidVolumeBuilder tests also use Be I decided for Be.